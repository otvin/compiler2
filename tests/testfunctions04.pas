program testfunctions04(output);

{test functions that call other functions}

function plustwo(q : Integer) : Integer;
begin
    plustwo := q +2
end;

function funcB(b : Integer) : Integer;
begin
    funcB := (b+2) * (b-4)
end;

function funcC(r: Integer) : Integer;
begin
    funcC := plustwo(r) * (r-4)
end;

function funcD(s: Integer) : Integer;
var
    tmp : Integer;
begin
    tmp := plusTwo(s);
    funcD := tmp * (s-4)
end;

begin {main program}

    writeln(funcB(8));
    writeln(plustwo(2));
    writeln(funcC(12));
    writeln(funcD(-7))

end.
