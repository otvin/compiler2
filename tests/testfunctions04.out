40
4
112
55
